pkg/languagedetection/internal/detectors/go_detector.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package detectors
+
+import (
+	"debug/elf"
+	"fmt"
+
+	dderrors "github.com/DataDog/datadog-agent/pkg/errors"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/network/go/binversion"
+)
+
+type GoDetector struct{}
+
+// DetectLanguage allows for detecting if a process is a go process, and it's version.
+// Note that currently the GoDetector only returns non-retriable errors since in all cases we will not be able to detect the language.
+// Scenarios in which we can return an error:
+//   - Invalid permissions. The system-probe is not running as root, or the container does not have CAP_PTRACE.
```

Comment:
That's not a real scenario 😄

Reply:
It's possible for the customer to misconfigure the system probe in this way, no? Especially the part where they need to run the system probe with `CAP_PTRACE`

Reply:
I think we'll hit permission error much sooner
anyway it is not critical to remove, just bare in mind that system-probe is running with high privileges and permissions so the different modules should not take any action to ensure that